### PR TITLE
CVE 2021 3538 fix

### LINKED ISSUE #1368: Replace github.com/satori/go.uuid (CVE-2021-3538)
Last release of https://github.com/satori/go.uuid is 1.2.0 from 2018 and contains CVE-2021-3538.

I'm not sure what an appropriate replacement would be. Maybe https://pkg.go.dev/gitee.com/gofrs/uuid like mentioned in https://github.com/satori/go.uuid/issues/118?

### PR DESCRIPTION
**Pull request by chaitanya-baraskar**
## Change Overview

Fix for CVE-2021-3538. Using `github.com/gofrs/uuid` instead of `github.com/satori/go.uuid` for uuid.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- Fixes #1368 

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Executed `make all-build` to build code.

### CODE REVIEW

File: pkg/blockstorage/awsefs/awsefs.go [hunk 1]
```diff
@@ -112,7 +112,8 @@ func (e *Efs) Type() blockstorage.Type {
 // volume info that is sent back from the AWS EFS.
 func (e *Efs) VolumeCreate(ctx context.Context, volume blockstorage.Volume) (*blockstorage.Volume, error) {
 	req := &awsefs.CreateFileSystemInput{}
-	req.SetCreationToken(uuid.NewV4().String())
+	reqId, _ := uuid.NewV4()
```
Comment (ankitjain235) [inline pkg/blockstorage/awsefs/awsefs.go:115]: We should not ignore the error here and at other places right?

Comment (ankitjain235) [inline pkg/blockstorage/awsefs/awsefs.go:115]: We should not ignore the error here and at other places right?

File: go.mod [hunk 2]
```diff
@@ -61,6 +60,8 @@ require (
 	sigs.k8s.io/controller-runtime v0.11.2
 )
 
+require github.com/gofrs/uuid v4.2.0+incompatible
```
Comment (julio-lopez) [inline go.mod:63]: Why no add this to the block above?

File: pkg/blockstorage/awsefs/awsefs.go (hunk 1 above)
Comment (julio-lopez) [reply pkg/blockstorage/awsefs/awsefs.go:115]: Correct. It should be checked and when not nil, wrap and return.
Same below.

File: go.mod (hunk 2 above)
Comment (pavannd1) [inline go.mod:63]: I think this should be added to the above list without the `+incompatible` part.

File: go.mod [hunk 3]
```diff
@@ -40,7 +40,6 @@ require (
 	github.com/openshift/client-go v0.0.0-20200521150516-05eb9880269c
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.12.1
-	github.com/satori/go.uuid v1.2.0
```
Comment (pavannd1) [inline go.mod:43]: As mentioned below, something like this
```
github.com/gofrs/uuid v4.2.0
```

File: pkg/blockstorage/awsefs/awsefs.go (hunk 1 above)
Comment (pavannd1) [reply pkg/blockstorage/awsefs/awsefs.go:115]: I agree. This is production code, we should not ignore errors.

File: go.mod (hunk 2 above)
Comment (julio-lopez) [reply go.mod]: the `+incompatible` is because it does not follow Go's guidelines for module versioning.

File: go.mod [hunk 4]
```diff
@@ -103,7 +102,7 @@ require (
 	github.com/go-openapi/jsonreference v0.19.5 // indirect
 	github.com/go-openapi/swag v0.19.14 // indirect
 	github.com/go-stack/stack v1.8.0 // indirect
-	github.com/gofrs/uuid v4.0.0+incompatible // indirect
+	github.com/gofrs/uuid v4.2.0+incompatible
```
Comment (julio-lopez) [inline go.mod:105]: This should go in the direct dependencies.

Has `go mod tidy -compat=1.17` been run? Does it change the file?

Comment (chaitanya-baraskar) [reply go.mod:105]: Let me check.

**Comment by pavannd1** [comment]
Reorganize the code a bit to reduce complexity - maybe split of logical modules into helper functions
```
pkg/blockstorage/azure/azuredisk.go:135:1: cognitive complexity 31 of func `(*AdStorage).SnapshotCopyWithArgs` is high (> 30) (gocognit)
func (s *AdStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.Snapshot, to blockstorage.Snapshot, args map[string]string) (*blockstorage.Snapshot, error) {
```

**Review by julio-lopez — APPROVED** [review]
Let's merge and iterate if needed

**Pull request merged**